manual/src/server/sal.md

```diff
@@ -0,0 +1,789 @@
+# SAL
+
+## Tfgrid
+
+### Load
+- loads key
+- chooses network
+- starts session and creates state for client
+
+```
+func (c *Client) Load(args Load) error
+```
+
+### DeployVM
+- Deploy a single VM 
+- Choose node id, farm id, etc 
+
+```
+func (c *Client) DeployVM(args tfgridBase.VMDeployment) (tfgridBase.VMDeployment, error)
+```
+
+### GetVMDeployment
+- Retrieve information about the deployed vm
+
+```
+func (c *Client) GetVMDeployment(name string) (tfgridBase.VMDeployment, error)
+```
+
+### CancelVMDeployment
+- Cancel the vm deployment
+
+```
+func (c *Client) CancelVMDeployment(name string) error
+```
+
+### DeployNetwork
+- Deploy VMs that will be in the same network
+- Provide name and network specs
+- Provide the VM specs (on what node, farmid, flist, etc.)
+
+```
+func (c *Client) DeployNetwork(args tfgridBase.NetworkDeployment) (tfgridBase.NetworkDeployment, error)
+```
+
+### GetNetworkDeployment
+- Retrieve information about the vms you deployed by providing the name when deploying
+```
+func (c *Client) GetNetworkDeployment(name string) (tfgridBase.NetworkDeployment, error)
+```
+
+### CancelNetworkDeployment
+- Delete all deployed VMs that are part of the model that goes by the modelName
+
+```
+func (c *Client) CancelNetworkDeployment(name string) error
+```
+
+### ExtendNetworkDeployment
```

Comment:
can't we use AttachNeworkToVm as a name?

Reply:
Hmm although it's a great name it would not reflect what the function actually does. The function doesn't attach a network to a vm, it creates a new vm in that network. Would AttachVmToNetwork be better?

Reply:
I thought it expands the network on ZOS node not a VM? @MarioBassem can you confirm please?

Reply:
i believe it's not called ExtendNetworkDeployemnt, or at least not in the current implementation.
the calls that achieve this are:
- https://github.com/threefoldtech/web3_proxy/blob/13a10a86572569dbe8b780c91a3f28a86d375489/server/pkg/tfgrid/client.go#L109
- https://github.com/threefoldtech/web3_proxy/blob/13a10a86572569dbe8b780c91a3f28a86d375489/server/pkg/tfgrid/client.go#L154
- https://github.com/threefoldtech/web3_proxy/blob/13a10a86572569dbe8b780c91a3f28a86d375489/server/pkg/tfgrid/solutions.go#L146

network is abstracted from users in the tfgrid client, so whenever a deployment is required, the client makes sure the node belongs to the right network

Reply:
Yes indeed it is those functions. We are reworking the structure and naming of it. But what does the MachinesAdd function do? It deploys a new VM to the network right?

Reply:
yes it does both things:
- add a ZOS network workload to the node if it's not already part of the network
- add a VM workload connected to the network